cookbook/knowledge/search_type/README.md

```diff
@@ -0,0 +1,254 @@
+# Search Types
+
+Search strategies determine how your agents find relevant information in knowledge bases. The right approach balances semantic understanding, keyword precision, and performance for optimal retrieval quality.
+
+## Core Concepts
+
+| Concept | Description |
+|---------|-------------|
+| **Vector Search** | Semantic similarity-based search using embeddings |
+| **Keyword Search** | Traditional text-based search using exact matches |
+| **Hybrid Search** | Combines vector and keyword search for better results |
+| **Search Type** | Configurable search algorithm in vector databases |
+
+## Getting Started
+
+### 1. Vector Search
+
+```python
+from agno.knowledge.knowledge import Knowledge
+from agno.vectordb.pgvector import PgVector, SearchType
+
+vector_db = PgVector(
+    table_name="recipes", 
+    db_url="postgresql+psycopg://ai:ai@localhost:5532/ai", 
+    search_type=SearchType.vector
+)
+knowledge = Knowledge(vector_db=vector_db)
+
+# Add content and search
+knowledge.add_content(url="https://agno-public.s3.amazonaws.com/recipes/ThaiRecipes.pdf")
+results = vector_db.search("chicken coconut soup", limit=5)
+```
+
+### 2. Keyword Search
+
+```python
+keyword_db = PgVector(
+    table_name="recipes", 
+    db_url="postgresql+psycopg://ai:ai@localhost:5532/ai", 
+    search_type=SearchType.keyword
+)
+
+results = keyword_db.search("chicken coconut soup", limit=5)
+```
+
+### 3. Hybrid Search
+
+```python
+hybrid_db = PgVector(
+    table_name="recipes", 
+    db_url="postgresql+psycopg://ai:ai@localhost:5532/ai", 
+    search_type=SearchType.hybrid
+)
+
+results = hybrid_db.search("chicken coconut soup", limit=5)
+```
+
+## Search Implementations
+
+### 1. Vector Search (`vector_search.py`)
```

Comment:
Rather use agents. We want to show that knowledge can be used standalone, but this makes it seem like it can ONLY be used standalone